ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1309,98 +1252,236 @@ private struct ThrottleState<Value> {
-/// `combineLatestWith`.
+
+extension SignalType {
+	private final func mapToAny() -> Signal<Any,Error> {
+		return self.map { $0 }
+	}
+}
 @warn_unused_result(message="Did you forget to call `observe` on the signal?")
-public func combineLatest<A, B, Error>(a: Signal<A, Error>, _ b: Signal<B, Error>) -> Signal<(A, B), Error> {
-	return a.combineLatestWith(b)
+public func combineLatest<A : SignalType, B : SignalType where A.Error == B.Error>(a: A, _ b: B) -> Signal<(A.Value, B.Value), A.Error> {
+	let anySignals: [Signal<Any,A.Error>] = [a.mapToAny(),b.mapToAny()]
+	return combineLatest(anySignals).map { $0.toTuple() }
 }
 
 /// Combines the values of all the given signals, in the manner described by
 /// `combineLatestWith`.
 @warn_unused_result(message="Did you forget to call `observe` on the signal?")
 public func combineLatest<A, B, C, Error>(a: Signal<A, Error>, _ b: Signal<B, Error>, _ c: Signal<C, Error>) -> Signal<(A, B, C), Error> {
-	return combineLatest(a, b)
-		.combineLatestWith(c)
-		.map(repack)
+	
+	let anySignals: [Signal<Any,Error>] =
+		[a.mapToAny(),
+		 b.mapToAny(),
+		 c.mapToAny()]
+	return combineLatest(anySignals).map { $0.toTuple() }
 }
 
 /// Combines the values of all the given signals, in the manner described by
 /// `combineLatestWith`.
 @warn_unused_result(message="Did you forget to call `observe` on the signal?")
 public func combineLatest<A, B, C, D, Error>(a: Signal<A, Error>, _ b: Signal<B, Error>, _ c: Signal<C, Error>, _ d: Signal<D, Error>) -> Signal<(A, B, C, D), Error> {
-	return combineLatest(a, b, c)
-		.combineLatestWith(d)
-		.map(repack)
+	
+	let anySignals: [Signal<Any,Error>] =
+		[a.mapToAny(),
+	 	 b.mapToAny(),
+		 c.mapToAny(),
+		 d.mapToAny()]
+	return combineLatest(anySignals).map { $0.toTuple() }
 }
 
+
 /// Combines the values of all the given signals, in the manner described by
 /// `combineLatestWith`.
 @warn_unused_result(message="Did you forget to call `observe` on the signal?")
 public func combineLatest<A, B, C, D, E, Error>(a: Signal<A, Error>, _ b: Signal<B, Error>, _ c: Signal<C, Error>, _ d: Signal<D, Error>, _ e: Signal<E, Error>) -> Signal<(A, B, C, D, E), Error> {
-	return combineLatest(a, b, c, d)
-		.combineLatestWith(e)
-		.map(repack)
+	
+	let anySignals: [Signal<Any,Error>] =
+		[a.mapToAny(),
+		 b.mapToAny(),
+		 c.mapToAny(),
+		 d.mapToAny(),
+		 e.mapToAny()]
+	return combineLatest(anySignals).map { $0.toTuple() }
 }
 
 /// Combines the values of all the given signals, in the manner described by
 /// `combineLatestWith`.
 @warn_unused_result(message="Did you forget to call `observe` on the signal?")
 public func combineLatest<A, B, C, D, E, F, Error>(a: Signal<A, Error>, _ b: Signal<B, Error>, _ c: Signal<C, Error>, _ d: Signal<D, Error>, _ e: Signal<E, Error>, _ f: Signal<F, Error>) -> Signal<(A, B, C, D, E, F), Error> {
-	return combineLatest(a, b, c, d, e)
-		.combineLatestWith(f)
-		.map(repack)
+	
+	let anySignals: [Signal<Any,Error>] =
```

Comment:
I don't doubt this doesn’t behave the same way, but the compiler is not making the same guarantees anymore due to the type erasure going on here.